sync/class.jetpack-sync-actions.php

```diff
@@ -119,12 +119,19 @@ static function set_is_importing_true() {
 	static function send_data( $data, $codec_name, $sent_timestamp, $queue_id ) {
 		Jetpack::load_xml_rpc_client();
 
-		$url = add_query_arg( array(
+		$query_args = array(
 			'sync'      => '1', // add an extra parameter to the URL so we can tell it's a sync action
 			'codec'     => $codec_name, // send the name of the codec used to encode the data
 			'timestamp' => $sent_timestamp, // send current server time so we can compensate for clock differences
 			'queue'     => $queue_id, // sync or full_sync
-		), Jetpack::xmlrpc_api_url() );
+		);
+
+		if ( Jetpack::sync_idc_optin() || Jetpack::is_development_version() ) {
```

Comment:
What about having the development version check as part of `sync_idc_optin` so the filter can be a final arbitration if they're opting in?